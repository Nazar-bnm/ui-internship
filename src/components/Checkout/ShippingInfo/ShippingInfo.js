import React from 'react';
import cx from 'classnames';

const CN = 'shipping-info';

class ShippingInfo extends React.Component {
  render() {
    return (
      <div className={cx(CN)}>
       ShippingInfo
      </div>
    );
  }
}

export default ShippingInfo;
