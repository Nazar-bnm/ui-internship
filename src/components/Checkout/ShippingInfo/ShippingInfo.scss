@import '../../../scss/global_styles';

.shipping-info {
  padding: 20px;
  font-family: font(primary-light);
  text-align: justify;
}