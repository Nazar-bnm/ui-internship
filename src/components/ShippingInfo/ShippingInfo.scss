@import '../../scss/global_styles';

.shippingInfo {
  justify-content: center;
  
  &__list {
    display: flex;
    justify-content: space-between;
    list-style: none;
    margin: 30px 0;
    width: 80%;

    &-item {
      display: flex;
      width: 30%;
    }
  }

  &__item {
    &-number {
      font-family: font(secondary);
      font-size: 1.4em;
    }

    &-text {
      &__wrapper {
        text-transform: uppercase;
        margin-left: 10px;
      }

      &__title {
        font-weight: 300;
        margin-bottom: 3px;
      }

      &__description {
        font-family: font(primary);
        font-size: 0.6em;
      }
    }
  }

  &__line {
    width: 100%;
    border: 0.5px solid $alto;
  }
}

@media screen and (max-width: 768px) {
  .shippingInfo__list {
    width: 100%;
  }
}

@media screen and (max-width: 480px) {
  .shippingInfo__list {
    &-item {
      flex-direction: column;
      text-align: center;
    }
  }

  .shippingInfo__item-text {
    &__wrapper {
      margin-left: 0;
      padding-top: 7px;
    }

    &__title {
      font-size: 0.8em;
    }
  }
}
