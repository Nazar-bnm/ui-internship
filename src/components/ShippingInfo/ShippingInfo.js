import React from 'react';
import cx from 'classnames';

import config from '../../../config';

import './ShippingInfo.scss';

const { shippingData } = config;
export const CN = 'shippingInfo';

const ShippingInfo = () => {
  const shippingInfo = shippingData.map(({ number, title, description }) => (
    <li key={number} className={`${CN}__list-item`}>
      <span className={`${CN}__item-number`}>
        0
        {number}
        .
      </span>
      <div className={`${CN}__item-text__wrapper`}>
        <h5 className={`${CN}__item-text__title`}>{title}</h5>
        <h6 className={`${CN}__item-text__description`}>{description}</h6>
      </div>
    </li>
  ));

  return (
    <div className={`content container ${cx(CN)}`}>
      <hr className={`${cx(CN)}__line`} />
      <ul className={`${cx(CN)}__list`}>
        {shippingInfo}
      </ul>
    </div>
  );
};

export default ShippingInfo;
