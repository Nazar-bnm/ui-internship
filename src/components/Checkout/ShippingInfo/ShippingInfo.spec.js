import React from 'react';
import { shallow } from 'enzyme';

import ShippingInfo from './ShippingInfo';

describe('<ShippingInfo />', () => {
  let wrapper = shallow(<ShippingInfo />);
  
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
