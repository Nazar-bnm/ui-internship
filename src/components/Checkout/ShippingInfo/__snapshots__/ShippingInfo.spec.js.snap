// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShippingInfo /> should match snapshot 1`] = `
<div
  className="shipping-info"
>
  Unless otherwise agreed, the delivery is done by mail. The product will be delivered exclusively to the address the customer placed in the order form. The customer has to ensure that the address information cited is complete and correct. Additional costs caused by incompleted information are to be replaced by the customer.
  <br />
  <br />
  Once your shipment has been sent, you will receive tracking information by e-mail.
  <br />
  <br />
  Delivery information are not binding unless a delivery period has been exceptionally confirmed.
</div>
`;
